bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/SolarwattHandlerFactory.java

```diff
@@ -0,0 +1,84 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.solarwatt.internal.channel.SolarwattChannelTypeProvider;
+import org.openhab.binding.solarwatt.internal.handler.DeviceHandler;
+import org.openhab.binding.solarwatt.internal.handler.EnergyManagerHandler;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolarwattHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Sven Carstens - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.solarwatt", service = ThingHandlerFactory.class)
+public class SolarwattHandlerFactory extends BaseThingHandlerFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(SolarwattHandlerFactory.class);
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_ENERGY_MANAGER,
+            THING_TYPE_INVERTER, THING_TYPE_POWERMETER, THING_TYPE_EVSTATION, THING_TYPE_BATTERYCONVERTER,
+            THING_TYPE_LOCATION, THING_TYPE_PVPLANT, THING_TYPE_GRIDFLOW);
+
+    private final HttpClient commonHttpClient;
+    private final SolarwattChannelTypeProvider channelTypeProvider;
+
+    @Activate
+    public SolarwattHandlerFactory(final @Reference HttpClientFactory httpClientFactory,
+            final @Reference SolarwattChannelTypeProvider channelTypeProvider) {
+        this.commonHttpClient = httpClientFactory.getCommonHttpClient();
+        this.channelTypeProvider = channelTypeProvider;
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (THING_TYPE_ENERGY_MANAGER.equals(thingTypeUID)) {
+            // energy manager is a separate device as it is the bridge
+            return new EnergyManagerHandler((Bridge) thing, this.channelTypeProvider, this.commonHttpClient);
+        } else if (this.supportsThingType(thingTypeUID)) {
+            // standard device handling
+            return new DeviceHandler(thing, this.channelTypeProvider);
+        }
+
+        this.logger.error("No matching handler for {} from {}", thingTypeUID, SUPPORTED_THING_TYPES_UIDS);
```

Comment:
should be warn and not error? Error is for critical problems that can cause openHAB to have issues.

See https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Reduced loglevels.